includes/Dashboard/Templates/Products.php

```diff
@@ -520,6 +545,83 @@ public function load_add_new_product_popup() {
         dokan_get_template_part( 'products/tmpl-add-product-popup' );
     }
 
+    /**
+     * Returns new category select ui html elements.
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return html
+     */
+    public function load_add_category_modal() {
+        /**
+         * Checking if current page is dokan dashboard.
+         * And
+         * Checking if it is a dokan add product page.
+         * Or
+         * Checking it is a edit page or product page.
+         */
+        if ( 'dashboard' === get_query_var('pagename','dokan-not-dashboard-page') && ( 'dokan-not-add-product-page' !== get_query_var( 'new-product', 'dokan-not-add-product-page' ) || 'dokan-not-products-page' != get_query_var( 'products', 'dokan-not-products-page' ) ) ) {
+            wp_enqueue_style( 'dokan-single-category-ui-css' );
+
+            $is_single = dokan_get_option( 'product_category_style', 'dokan_selling', 'single' ) == 'single' ? true : false;
+
+            $args = array(
+                'taxonomy'     => 'product_cat',
+                'orderby'      => 'name',
+                'show_count'   => 1,      // 1 for yes, 0 for no
+                'pad_counts'   => 1,      // 1 for yes, 0 for no
+                'hierarchical' => 1,      // 1 for yes, 0 for no
+                'title_li'     => '',
+                'hide_empty'   => 0,
+            );
+            $all_categories = get_categories( $args );
+
+            foreach ( $all_categories as $key => $value ) {
+                $children = get_terms( 'product_cat', array(
+                    'parent'    => $value->term_id,
+                    'hide_empty' => false
+                ) );
+
+                $children ? $all_categories[$key]->has_child = true : $all_categories[$key]->has_child = false;
```

Comment:
Here also the basic phpcs is violated. It should be `$all_categories[ $key ]`. Please fix all the phpcs first. Only after that the review will resume.